### PR TITLE
Chore/co 326 remove imapd

### PR DESCRIPTION
**Pull request by frisonisland**
### Goal
Remove imapd.
Imap can be handled in two ways: in-mailbox, with mailboxd, or in a standalone process with imapd.

Related: https://github.com/Zextras/carbonio-core-utils/pull/18

### What's changed

- ImapDaemon class has been removed.
- Other classes related to Imapd have been removed:

  * imapd statistics trackers and exporter
  * flushCache for imapd servers
  * account logger on imapd

- Imapd related attributes (4 of them) have been deprecated, and getters have been generated (except zimbraAuthFallbackToLocal to avoid regression).
- finally some ant tasks related to imapd management have been removed

### CODE REVIEW

**Review by keshavbhatt — CHANGES_REQUESTED** [review]
Dropped a few comments,
Please check if we still require store-conf/conf/imapd.log4j.properties after these changes. 

Everything else LGTM.

File: soap/src/java/com/zimbra/soap/admin/type/CacheSelector.java [hunk 1]
```diff
@@ -41,18 +41,6 @@ public final class CacheSelector {
...
-  /**
-   * @zm-api-field-tag imap-servers
-   * @zm-api-field-description
-   *     <table>
-   * <tr> <td> <b>0 (false)</b> </td> <td> don't issue X-ZIMBRA-FLUSHCACHE IMAP command to upstream IMAP servers </td> </tr>
-   * <tr> <td> <b>1 (true) [default]</b> </td>
-   *      <td> flush cache on servers listed in zimbraReverseProxyUpstreamImapServers for the current server via X-ZIMBRA-FLUSHCACHE</td> </tr>
-   * </table>
-   */
-  @XmlAttribute(name = AdminConstants.A_IMAPSERVERS /* imapServers */, required = false)
```
[the remark this answers is not in the document]
Comment (frisonisland) [reply soap/src/java/com/zimbra/soap/admin/type/CacheSelector.java:53]: Ty @keshavbhatt , I removed after double-checking it was unused. 
Also remove imapd.log4j.properties as it was loaded inside ImapDaemon class and that has been removed.

**Review by keshavbhatt — APPROVED** [review]

**Review by Polpetta — APPROVED** [review]
Is it a big problem to add `@Deprecated(since= "22.9.0")` on top of methods?

**Comment by frisonisland** [comment]
> Is it a big problem to add `@Deprecated(since= "22.9.0")` on top of methods?

I'm on you with this, but that code is autogenerated and atm it's like that :/. 
We can take care about when we will improve auto-generators

**Pull request merged**